editor/editor_settings.cpp

```diff
@@ -1485,6 +1486,26 @@ bool EditorSettings::is_default_text_editor_theme() {
 	return _is_default_text_editor_theme(p_file.get_file().to_lower());
 }
 
+void EditorSettings::migrate_settings() {
+	// Migrate the setting Always Open Output On Play that was True by default.
+	// If false, then switch Action On Play to "Do Nothing" to keep same behavior.
+	if (has_setting("run/output/always_open_output_on_play")) {
+		if (!bool(get("run/output/always_open_output_on_play"))) {
+			set("run/output/action_on_play", 1);
+			erase("run/output/always_open_output_on_play");
+		}
+	}
+
+	// Migrate the setting Always Close Output On Stop that was False by default.
+	// If true, then switch Action On Stop to "Close Bottom Panel" to keep same behavior.
+	if (has_setting("run/output/always_close_output_on_stop")) {
+		if (bool(get("run/output/always_close_output_on_stop"))) {
+			set("run/output/action_on_stop", 0);
+			erase("run/output/always_close_output_on_stop");
+		}
+	}
+}
```

Comment:
Since this is compatibility code, we would usually make it conditional to this define:
```suggestion
#ifndef DISABLE_DEPRECATED
void EditorSettings::migrate_settings() {
	// Migrate the setting Always Open Output On Play that was True by default.
	// If false, then switch Action On Play to "Do Nothing" to keep same behavior.
	if (has_setting("run/output/always_open_output_on_play")) {
		if (!bool(get("run/output/always_open_output_on_play"))) {
			set("run/output/action_on_play", 1);
			erase("run/output/always_open_output_on_play");
		}
	}

	// Migrate the setting Always Close Output On Stop that was False by default.
	// If true, then switch Action On Stop to "Close Bottom Panel" to keep same behavior.
	if (has_setting("run/output/always_close_output_on_stop")) {
		if (bool(get("run/output/always_close_output_on_stop"))) {
			set("run/output/action_on_stop", 0);
			erase("run/output/always_close_output_on_stop");
		}
	}
}
#endif
```

Likewise in the header, and the place where it's called.

Reply:
Actually though, I'm not sure we should bother migrating those settings:
- We've changed a lot of settings in the past and never migrated. Users for the most part seem to accommodate fine with having to retweak a few settings when upgrading.
- Keeping the old settings means that the configuration still works on older editor versions. This isn't as relevant following #90875 though in this particular context of making this change in 4.3 (but if we were to backport this to 4.2, it would be relevant).
- Since we add new options, this is an opportunity for users to reconsider their preferred option and maybe discover a behavior that will improve their experience.

Reply:
I removed the setting migration. Clearly it was professional deformation on my part, I developed management systems for the last 25 years and migration settings was always essential! It will be removed in my next commit.
I like your thinking!